src/ssr.rs

```diff
@@ -1,156 +1,187 @@
-use std::collections::HashMap;
-
-#[derive(Clone, Debug, PartialEq)]
-pub struct Ssr<'a> {
-    // TODO: Check if better Box<str> instead of String
-    source: String,
-    entry_point: &'a str,
+pub struct SSREnvironment {
+    source_code: String,
+    entry_point: String,
+    root_export: String,
```

Comment:
Why do we keep in memory all the inputs? Couldn't we export just the function (line: 71) and the scope in order to just pass the `params` at each user request? Doing so we will keep the unsafe scope just at during the start and not during the process.

Reply:
We cannot keep both the `scope` and the `isolate` in the same struct because the `scope` refers to the `isolate`. This would lead to a self referential struct. This was the direction I went with until I realized that self referential structs are a pain to get going with safe Rust.

Reply:
could it be possible keep just the `scope` without the `isolate`?

Reply:
If we don't move the Isolate into the struct then it gets dropped at the end of the constructor and scope isn't valid anymore.